pkg/controller/wip/controller_test.go

```diff
@@ -96,59 +134,113 @@ func TestReconcileBroker(t *testing.T) {
 		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
 	}
 
-	close(stopCh)
 }
 
-func TestReconcileInstanceHappyLand(t *testing.T) {
-	fakeKubeClient, fakeCatalogClient, fakeBrokerCatalog, _, _, testController, sharedInformers, stopCh := newTestController(t)
+func TestReconcileInstanceNonExistentServiceClass(t *testing.T) {
+	_, fakeCatalogClient, _, _, _, testController, _, stopCh := newTestController(t)
+	defer close(stopCh)
 
-	fakeBrokerCatalog.RetCatalog = &brokerapi.Catalog{
-		Services: []*brokerapi.Service{
-			{
-				Name:        "test-service",
-				ID:          "12345",
-				Description: "a test service",
-				Plans: []brokerapi.ServicePlan{
-					{
-						Name:        "test-plan",
-						Free:        true,
-						ID:          "34567",
-						Description: "a test plan",
-					},
-				},
-			},
+	instance := &v1alpha1.Instance{
+		ObjectMeta: v1.ObjectMeta{Name: "test-instance"},
+		Spec: v1alpha1.InstanceSpec{
+			ServiceClassName: "nothere",
+			PlanName:         "nothere",
+			OSBGUID:          instanceGUID,
 		},
 	}
 
-	broker := &v1alpha1.Broker{
-		ObjectMeta: v1.ObjectMeta{Name: "test-broker"},
-		Spec: v1alpha1.BrokerSpec{
-			URL:     "https://example.com",
-			OSBGUID: "OSBGUID field",
+	testController.reconcileInstance(instance)
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 1, len(actions); e != a {
+		t.Logf("%+v\n", actions)
```

Comment:
I don't think we need the log here. it would be better if this info were part of the Fatal msg so that it shows up on the testing output.